### PR TITLE
Cleanup connection class

### PR DESCRIPTION
**Pull request by RobertMe**
These changes don't change any behavior, but they do clean up a lot of old/unused and duplicate code in the connection class. So the main purpose is to make it easier to maintain the class.

NOTE: Contains commit of #1 so same warning applies. These changes can be reviewed, but please hold of merging until I've figured out what is the best way to do the rename.

### CODE REVIEW

File: libkodimote/kodiconnection.cpp [hunk 1]
```diff
@@ -259,10 +247,24 @@ void KodiConnectionPrivate::slotConnected()
...
+    if(m_kodiVersionMajor < 6) {
+        qDebug() << "WARNING! XBMC is too old or version field not valid! Some features might not work";
+        m_connectionError = tr("This version of Kodimote is designed to work with XBMC Frodo (v12.0). It seems you have connected to an older version of XMBC. Please upgrade XBMC in order to use Kodimote.");
+    } else {
+        m_connected = true;
+        m_connectionError.clear();
+    }
+    m_connecting = false;
+    emit m_notifier->connectionChanged();
+    m_host->setPersistent(true);
```
Comment (mzanetti) [inline libkodimote/kodiconnection.cpp]: I think the host should only be stored when actually connected successfully.

**Comment by mzanetti** [comment]
There are some changes in the sailfish package files which I think are duplicated with another branch

**Comment by RobertMe** [comment]
Yep. Thats what the note is about. Without those I couldn't build.

Will rebase when they can build on master (+ fixing above + gitorious issues)

File: libkodimote/kodiconnection.cpp (hunk 1 above)
Comment (RobertMe) [inline libkodimote/kodiconnection.cpp]: Behavior is the same as current master: https://github.com/mzanetti/kodimote/blob/21b60c0544702498acbe1942003ed5cfe25fece5/libkodimote/kodiconnection.cpp#L407

This code is only ran when:
1. The socket is opened/connected (as it is called from slotConnected)
2. The HTTP calls work (as the version command runs over HTTP)
So this is already done when you are successfully connected.

**Comment by RobertMe** [comment]
Rebased. Addressed issues of https://gitorious.org/xbmcremote/xbmcremote/merge_requests/38

It's now independent of #1 (and thus can be merged), but Sailfish app isn't directly runnable from the IDE (needs either #1 or a change in the sailfish.pro file (changing TARGET back to harbour-xbmcremote)).

**Comment by mzanetti** [comment]
crashes when switching between connections

**Comment by RobertMe** [comment]
Could you please try with #15 applied? I would guess it's related to that (as the playlists get reloaded when reconnecting)

**Pull request merged**